Dockerfile

```
@@ -1,149 +1,207 @@
+                    --with-cxx-std-min=2014 \
+        && make 2>/dev/null -j${compile_cores} install \
+        && rm -rf ${LIBMESH_INSTALL_DIR}/build ${LIBMESH_INSTALL_DIR}/libmesh ; \
     fi
 
-# Clone and install MOAB
-RUN if [ "$include_dagmc" = "true" ] ; \
-    then pip install --upgrade numpy cython ; \
-    mkdir MOAB ; \
-    cd MOAB ; \
-    mkdir build ; \
-    git clone  --single-branch --branch 5.2.1 https://bitbucket.org/fathomteam/moab.git ; \
-    cd build ; \
-    cmake ../moab -DENABLE_HDF5=ON \
-                  -DENABLE_NETCDF=ON \
-                  -DBUILD_SHARED_LIBS=OFF \
-                  -DENABLE_FORTRAN=OFF \
-                  -DENABLE_BLASLAPACK=OFF ; \
-    make -j"$compile_cores" ; \
-    make -j"$compile_cores" install ; \
-    rm -rf * ; \
-    cmake ../moab -DBUILD_SHARED_LIBS=ON \
-                  -DENABLE_HDF5=ON \
-                  -DENABLE_PYMOAB=ON \
-                  -DENABLE_FORTRAN=OFF \
-                  -DENABLE_BLASLAPACK=OFF ; \
-    make -j"$compile_cores" ; \
-    make -j"$compile_cores" install ; \
-    fi
-
-# Clone and install Double-Down
-RUN if [ "$include_dagmc" = "true" ] ; \
-    then git clone --single-branch --branch main https://github.com/pshriwise/double-down.git ; \
-    cd double-down ; \
-    mkdir build ; \
-    cd build ; \
-    cmake .. -DCMAKE_INSTALL_PREFIX=.. \
-             -DMOAB_DIR=/usr/local \
-             -DEMBREE_DIR=/embree ; \
-    make -j"$compile_cores" ; \
-    make -j"$compile_cores" install ; \
-    fi
-
-# Clone and install DAGMC
-RUN if [ "$include_dagmc" = "true" ] ; \
-    then mkdir DAGMC ; \
-    cd DAGMC ; \
-    git clone --single-branch --branch 3.2.0 https://github.com/svalinn/DAGMC.git ; \
-    mkdir build ; \
-    cd build ; \
-    cmake ../DAGMC -DBUILD_TALLY=ON \
-                   -DCMAKE_INSTALL_PREFIX=/DAGMC/ \
-                   -DMOAB_DIR=/usr/local \
-                   -DBUILD_STATIC_LIBS=OFF \
-                   -DBUILD_STATIC_EXE=OFF ; \
-    make -j"$compile_cores" install ; \
-    rm -rf /DAGMC/DAGMC /DAGMC/build ; \
-    fi
-
-# Clone and install OpenMC with DAGMC
-RUN if [ "$include_dagmc" = "true" ] ; \
-    then git clone --recurse-submodules https://github.com/openmc-dev/openmc.git /opt/openmc ; \
-    cd /opt/openmc ; \
-    mkdir build ; \
-    cd build ; \
-    cmake -Doptimize=on \
-          -Ddagmc=ON \
-          -DDAGMC_DIR=/DAGMC/ \
-          -DHDF5_PREFER_PARALLEL=on ..  ; \
-    make -j"$compile_cores" ; \
-    make -j"$compile_cores" install ; \
-    cd ..  ; \
-    pip install -e .[test] ; \
-    fi
+# clone and install openmc
+RUN mkdir -p ${HOME}/OpenMC && cd ${HOME}/OpenMC \
+    && git clone --shallow-submodules --recurse-submodules -b ${openmc_branch} --depth=1 ${OPENMC_REPO} \
+    && mkdir build && cd build ; \
+    if [ ${build_dagmc} = "on" ] && [ ${build_libmesh} = "on" ]; then \
+        cmake ../openmc \
+            -Doptimize=on \
+            -Ddebug=on \
```

Setting both Doptimize and Ddebug to off explicitly just to be clear so anyone modifying the dockerfile can easily change them to on if they want to. What do you think, is this ok @gridley 

```suggestion
            -Doptimize=off \
            -Ddebug=off \
```

---

Seems reasonable to me! Makes sense to leave it open for modification.

---

If this is the way to go then I should do this for all 4 combination, but I can do this all in a single commit instead of this review comment method

---

Yeh that'd be the ticket, no doubt :D